### ISSUE TITLE
MinSize not respected using wx backend causes wxAssertionError. Bug fix included.

### ISSUE THREAD

**Issue by jmoraleda** [matplotlib/matplotlib#14883]
**Summary** 
Two problems occurs when the user manually resizes to a very small size a frame containing a figure.

**Background**

When a `FigureCanvasWxAgg` widget is added to a sizer using the `wx.EXPAND` flag, with a border, the `_onSize` method of `_FigureCanvasWxBase ` attempts to size the figure dynamically based on the available client area in the parent window. This is the expected behavior, and it works correctly if there is sufficient area for the figure size specified.

**Problem**

However if there is insufficient area, the size reported to the sizer becomes arbitrarily small. This is is not the correct or expected behavior:

- The user expects a control to never be drawn smaller than its `MinSize`: If there is insufficient size, it is the expected behavior that the control should extend beyond the parent window limits rather than become arbitrarily small.
- As the parent client area shrinks, the reported control size eventually has have a dimension equal to 0, which is invalid and yields the following error from the wx library:

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py", line 844, in _onSize
    self.bitmap = wx.Bitmap(self._width, self._height)
wx._core.wxAssertionError: C++ assertion ""width > 0 && height > 0"" failed at /home/jorge/vc/Phoenix/ext/wxWidgets/src/gtk/bitmap.cpp(540) in Create(): invalid bitmap size
```

**Code for reproduction**

```python
import matplotlib
matplotlib.use('WXAgg')
from matplotlib.figure import Figure
from matplotlib.backends.backend_wxagg import FigureCanvasWxAgg as FigureCanvas
import wx


class MyFrame(wx.Frame):
    def __init__(self, parent):
        super(MyFrame, self).__init__(parent)
        sizer = wx.BoxSizer(wx.VERTICAL)
        self.fig = Figure(figsize=(2,1))
        self.canvas = FigureCanvas(self, -1, self.fig)
        sizer.Add(self.canvas, 1, wx.EXPAND|wx.ALL, 5)
        self.SetSizer(sizer)

app = wx.App()
frm = MyFrame(None)
frm.Show()
app.MainLoop()
```

To trigger the error, manually resize the window that opens to a very small size by dragging its handles.

**BUG FIX**
To fix both symptons of the bug it suffices to ensure the size reported is always at least MinSize. This can be achieved by adding one line to the code. In particular, replacing the current block of code inside the `_onSize` method of `_FigureCanvasWxBase `
```python
        if sz and si and not si.Proportion and not si.Flag & wx.EXPAND:
            # managed by a sizer, but with a fixed size
            size = self.GetMinSize()
        else:
            # variable size
            size = self.GetClientSize()
```
with the following one
```python
        if sz and si and not si.Proportion and not si.Flag & wx.EXPAND:
            # managed by a sizer, but with a fixed size
            size = self.GetMinSize()
        else:
            # variable size
            size = self.GetClientSize()
            size.IncTo(self.GetMinSize()) # Do not allow size to become smaller than MinSize
```

**Comment by tacaswell** [comment]
Thanks for reporting this @jmoraleda !  Could you open a pull request with that change?

**Comment by jmoraleda** [comment]
Thank you for your very fast review @tacaswell. I just opened the pull request: https://github.com/matplotlib/matplotlib/pull/14884

**Comment by dstansby** [comment]
Fixed by #14884
